### PR TITLE
Sponsor match orders

### PR DESCRIPTION
**Pull request by jeremyjams**
In scope
- New unit tests covering sponsor match orders :heavy_check_mark: 

Out scope
- New integration test featuring a match orders sponsoring :next_track_button:

### CODE REVIEW

File: contracts/libs/IexecLibCore_v5.sol [hunk 1]
```diff
@@ -1,5 +1,6 @@
+// SPDX-FileCopyrightText: 2020-2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
 // SPDX-License-Identifier: Apache-2.0
-
+// TODO: Remove header bellow
```
Comment (jeremyjams) [inline contracts/libs/IexecLibCore_v5.sol:3]: Will be removed in some other next PRs if fine for you

File: test/000_fullchain-boost.test.ts [hunk 2]
```diff
@@ -71,7 +72,10 @@ async function extractRegistryEntryAddress(
 ): Promise<string> {
     const events = extractEventsFromReceipt(receipt, registryInstanceAddress, 'Transfer');
     if (events && events[0].args) {
-        const lowercaseAddress = events[0].args['tokenId'].toHexString();
+        const lowercaseAddress = ethers.utils.hexZeroPad(
+            BigNumber.from(events[0].args['tokenId']).toHexString(),
+            20,
+        );
```
Comment (jeremyjams) [inline test/000_fullchain-boost.test.ts:78]: Fix for:
`Error: invalid address (argument="address", value="0x7a96932ea86a8a23f336f4800f404e6361d42b", code=INVALID_ARGUMENT, version=address/5.7.0)` occurring some times.

File: contracts/modules/delegates/IexecPoco1Delegate.sol [hunk 3]
```diff
@@ -61,13 +61,85 @@ contract IexecPoco1Delegate is IexecPoco1, DelegateBase, IexecEscrow, SignatureV
...
+            _requestorder,
+            sponsor
+        );
+        return dealId;
+    }
+
+    /**
+     * Sponsor match orders for a requester.
+     * Unlike the standard `matchOrders(..)` hook where the requester pays for
+     * the deal, this current hook makes possible to any `msg.sender` to pay for
```
Comment (zguesmi) [inline contracts/modules/delegates/IexecPoco1Delegate.sol:93]:
```suggestion
     * the deal, this current hook makes it possible for any `msg.sender` to pay for
```

**Review by zguesmi — APPROVED** [review]

File: contracts/modules/delegates/IexecPoco1Delegate.sol [hunk 4]
```diff
@@ -61,13 +61,84 @@ contract IexecPoco1Delegate is IexecPoco1, DelegateBase, IexecEscrow, SignatureV
...
+                _apporder,
+                _datasetorder,
+                _workerpoolorder,
+                _requestorder,
+                _requestorder.requester
+            );
+    }
+
+    /**
+     * Sponsor match orders for a requester.
```
Comment (jeremyjams) [inline contracts/modules/delegates/IexecPoco1Delegate.sol:90]: @Amxx FYI

**Review by gfournieriExec — COMMENTED** [review]
Thank you Jérémy !

**Review by gfournieriExec — APPROVED** [review]

**Pull request merged**